src/api/participants/handler.js

```diff
@@ -102,9 +102,18 @@ const create = async function (request, h) {
         }
       }
     } else {
-      const participantCurrencyId1 = await ParticipantService.createParticipantCurrency(participant.participantId, request.payload.currency, ledgerAccountTypes.POSITION, false)
-      const participantCurrencyId2 = await ParticipantService.createParticipantCurrency(participant.participantId, request.payload.currency, ledgerAccountTypes.SETTLEMENT, false)
-      participant.currencyList = [await ParticipantService.getParticipantCurrencyById(participantCurrencyId1), await ParticipantService.getParticipantCurrencyById(participantCurrencyId2)]
+      const defaultSettlementModels = allSettlementModels.filter(model => model.currencyId === null)
+      if (Array.isArray(defaultSettlementModels) && defaultSettlementModels.length > 0) {
+        for (const settlementModel of defaultSettlementModels) {
+          const [participantCurrencyId1, participantCurrencyId2] = await Promise.all([
+            ParticipantService.createParticipantCurrency(participant.participantId, request.payload.currency, settlementModel.ledgerAccountTypeId, false),
```

Comment:
What happens if `createParticipantCurrency(...)` throws an error? I don't see a test case for that below.